relayer/workers/parachaincommitmentrelayer/beefy-listener.go

```diff
@@ -278,56 +265,61 @@ func createParachainHeaderProof(allParaHeads []types.Header, ourParaHead types.H
-	chainTypes.ChannelID,
-	types.H256,
-	[]chainTypes.CommitmentMessage,
-	error) {
+func (li *BeefyListener) extractCommitments(
+	header types.Header,
+	mmrProof types.GenerateMMRProofResponse,
+	ourParaHeadProof string) ([]MessagePackage, error) {
 
 	li.log.WithFields(logrus.Fields{
 		"blockNumber": header.Number,
 	}).Debug("Extracting commitment from parachain header")
 
-	digestItem, err := getAuxiliaryDigestItem(header.Digest)
+	auxDigestItems, err := getAuxiliaryDigestItems(header.Digest)
 	if err != nil {
-		return chainTypes.ChannelID{}, types.H256{}, nil, err
-	}
-
-	if digestItem == nil || !digestItem.IsCommitment {
-		return chainTypes.ChannelID{}, types.H256{}, nil, nil
+		return nil, err
 	}
 
-	li.log.WithFields(logrus.Fields{
-		"block":          header.Number,
-		"channelID":      digestItem.AsCommitment.ChannelID,
-		"commitmentHash": digestItem.AsCommitment.Hash.Hex(),
-	}).Debug("Found commitment hash in header digest")
-
-	channelID := digestItem.AsCommitment.ChannelID
-	commitmentHash := digestItem.AsCommitment.Hash
-	commitmentMessages, err := li.getMessagesForDigestItem(digestItem)
-	if err != nil {
-		return chainTypes.ChannelID{}, types.H256{}, nil, err
+	var messagePackages []MessagePackage
+	for _, auxDigestItem := range auxDigestItems {
+		li.log.WithFields(logrus.Fields{
+			"block":          header.Number,
+			"channelID":      auxDigestItem.AsCommitment.ChannelID,
+			"commitmentHash": auxDigestItem.AsCommitment.Hash.Hex(),
+		}).Debug("Found commitment hash in header digest")
+		commitmentHash := auxDigestItem.AsCommitment.Hash
+		commitmentMessagesData, err := li.getMessagesDataForDigestItem(&auxDigestItem)
+		if err != nil {
+			return nil, err
+		}
+		messagePackage := MessagePackage{
+			auxDigestItem.AsCommitment.ChannelID,
+			commitmentHash,
+			commitmentMessagesData,
+			ourParaHeadProof,
+			mmrProof,
+		}
+		messagePackages = append(messagePackages, messagePackage)
 	}
 
-	return channelID, commitmentHash, commitmentMessages, nil
+	return messagePackages, nil
 }
 
-func getAuxiliaryDigestItem(digest types.Digest) (*chainTypes.AuxiliaryDigestItem, error) {
+func getAuxiliaryDigestItems(digest types.Digest) ([]chainTypes.AuxiliaryDigestItem, error) {
+	var auxDigestItems []chainTypes.AuxiliaryDigestItem
 	for _, digestItem := range digest {
 		if digestItem.IsOther {
 			var auxDigestItem chainTypes.AuxiliaryDigestItem
 			err := types.DecodeFromBytes(digestItem.AsOther, &auxDigestItem)
 			if err != nil {
 				return nil, err
 			}
-			return &auxDigestItem, nil
+			auxDigestItems = append(auxDigestItems, auxDigestItem)
 		}
 	}
-	return nil, nil
+	return auxDigestItems, nil
 }
 
-func (li *BeefyListener) getMessagesForDigestItem(digestItem *chainTypes.AuxiliaryDigestItem) ([]chainTypes.CommitmentMessage, error) {
+func (li *BeefyListener) getMessagesDataForDigestItem(digestItem *chainTypes.AuxiliaryDigestItem) (types.StorageDataRaw, error) {
```

Comment:
Nit: There's no concept of a message in this scope.

```suggestion
func (li *BeefyListener) getDataForDigestItem(digestItem *chainTypes.AuxiliaryDigestItem) (types.StorageDataRaw, error) {
```

Reply:
will do